templates/infrastructure/environment/aws/eks/terragrunt.hcl

```diff
@@ -68,7 +68,26 @@ inputs = {
     "aws.amazon.com/spot": "true"
   }
 
-  {{if .Spec.ami_type}}ami_type = "{{.Spec.ami_type}}"{{end}}
+  node_groups = [{{ range $n := .Spec.node_groups }}
+    {
+      ng_name = "{{$n.ng_name}}"
+      desired_capacity = {{$n.number_of_instance}}
+      max_capacity = {{$n.number_of_instance_max}}
+      min_capacity = {{$n.number_of_instance_min}}
+      disk_size = {{$n.disk_size}}
+      instance_type = "{{$n.instance_type}}"
+      spot = {{$n.spot}}
+      ami_type = "{{$n.ami_type}}"
```

Comment:
where are we getting the ami type from? this would be different for each nodegroup architecture
arm, x86-64, GPU based etc.

Reply:
At the moment, the ami id is selected based on instanceype and passed in as input along with it. CC: @prajjwaldimri correct me if i am wrong here

Reply:
Ami Type is resolved by midard based on the instance type and sent to tf-modules

Reply:
What's the midgard logic? Ping me plz